AEPMessaging/Sources/FeedRulesEngine.swift

```diff
@@ -35,30 +35,29 @@ class FeedRulesEngine {
 
     /// if we have rules loaded, then we simply process the event.
     /// if rules are not yet loaded, add the event to the waitingEvents array to
-    func evaluate(event: Event) -> [String: Feed]? {
+    func evaluate(event: Event) -> [Surface: [Inbound]]? {
         let consequences = launchRulesEngine.evaluate(event: event)
         guard let consequences = consequences else {
             return nil
         }
 
-        var feeds: [String: Feed] = [:]
+        var inboundMessages: [Surface: [Inbound]] = [:]
         for consequence in consequences {
-            let details = consequence.details as [String: Any]
+            let detail = consequence.details as [String: Any]
 
-            if
-                let mobileParams = details[MessagingConstants.Event.Data.Key.FEED.MOBILE_PARAMETERS] as? [String: Any],
-                let feedItem = FeedItem.from(data: mobileParams, id: consequence.id)
-            {
-                let surfacePath = feedItem.surface ?? ""
+            guard let inboundMessage = Inbound.from(consequenceDetail: detail, id: consequence.id) else {
+                continue
+            }
+
+            let surfaceUri = inboundMessage.meta?[MessagingConstants.Event.Data.Key.Feed.SURFACE] as? String ?? ""
+            let surface = Surface(uri: surfaceUri)
```

Comment:
what's the correct thing to do here if we don't have a surface?  seems like maybe we ought to just `continue` if we don't find the surface in `inboundMessage.meta`

Reply:
Yes, that's why I checked if we could add some meta info by default e.g. surfaceUri in the inbound message and looks like that's possible.